memsource/memsource.py

```diff
@@ -7,11 +7,11 @@ def __init__(self, user_name=None, password=None, token=None, headers=None):
         If token is given, use the token.
         Otherwise authenticate with user_name and password, and get token.
         """
-        self.auth = api.Auth(token, headers)
         if user_name and password and not token and not headers:
-            token = self.auth.login(user_name, password).token
+            token = api.Auth(token, headers).login(user_name, password).token
```

Comment:
```suggestion
            token = api.Auth().login(user_name, password).token
```
`token` and `headers` will always be None in this case and will be replaced further down.

Reply:
Right @DarylPenetrante I'll just remove these since they are optional. Also the goal of this logic is just to call the login method.